src/lib/index.ts

```diff
@@ -97,39 +99,49 @@ export class GoogleToken {
 
     if (!this.key && this.keyFile) {
       const mimeType = mime.getType(this.keyFile);
-      if (mimeType === 'application/json') {
-        // json file
-        const key = await readFile(this.keyFile, 'utf8');
-        const body = JSON.parse(key);
-        this.key = body.private_key;
-        this.iss = body.client_email;
-        if (!this.key || !this.iss) {
-          const e = new Error('private_key and client_email are required.');
-          (e as NodeJS.ErrnoException).code = 'MISSING_CREDENTIALS';
-          throw e;
-        }
-      } else {
-        // Must be a .p12 file or .pem key
-        if (!this.iss) {
-          const e = new Error('email is required.');
-          (e as NodeJS.ErrnoException).code = 'MISSING_CREDENTIALS';
-          throw e;
-        }
-
-        if (mimeType === 'application/x-pkcs12') {
-          // convert to .pem on the fly
-          const key = await toPem(this.keyFile);
-          this.key = key;
-        } else {
-          // assume .pem key otherwise
-          const key = await readFile(this.keyFile, 'utf8');
-          this.key = key;
-        }
+      switch (mimeType) {
+        case 'application/json':
+          // *.json file
+          const key1 = await readFile(this.keyFile, 'utf8');
```

Comment:
Nit. If you embrace the `case` statement with parens, you can avoid naming local vars like `key1`, `key2`, etc.

```typescript
case 'application/json': {
  const key = ...
  ...
  break;
}
case 'application/x-x509-ca-cert': {
  const key = ...
  ...
}
...
```

Reply:
Thank you.  Hadn't had my coffee yet, and was wondering why it was complaining about the scope.